toucan_connectors/snowflake_oauth2/snowflake_oauth2_connector.py

```diff
@@ -203,6 +205,20 @@ def close_function(conn: SnowflakeConnection) -> None:
 
         return connection
 
+    def get_identifier(self):
+        json_uid = JsonWrapper.dumps(
+            {
+                'name': self.name,
+                'account': self.account,
+                'client_id': self.client_id,
+                'scope': self.scope,
+                'role': self.role,
+            },
+            sort_keys=True,
+        )
+        string_uid = str(uuid.uuid3(uuid.NAMESPACE_OID, json_uid))
+        return string_uid
```

Comment:
The idea for me here is good... but instead of passing the whole json string object to `uuid3`, i suggest to pass only a hash of the json string of the object, because following the [documentation](https://docs.python.org/3/library/uuid.html#uuid.uuid3) it should be only a `name` not a huge json string with a lot of elements in it, so am proposing this:

```python
# at the top of the file i import a builtin function
# md5 from hashlib / instead of uuid
from hashlib import md5

# ... some code here ...

# and i will structure the function in this way...
def get_identifier(self) -> str:
    # we get the hash of the json string
    json_uid = md5(JsonWrapper.dumps(
        {
            'name': self.name,
            'account': self.account,
            'client_id': self.client_id,
            'scope': self.scope,
            'role': self.role,
        },
        sort_keys=True,
    ).encode("utf-8")).hexdigest()
    
    return str(uuid.uuid3(uuid.NAMESPACE_OID, json_uid))
```